modules/ingestor/src/service/format.rs

```diff
@@ -184,6 +214,20 @@ where
     .await??)
 }
 
+async fn yaml_from_stream<S, T>(stream: S) -> Result<T, Error>
+where
+    S: Stream<Item = Result<Bytes, anyhow::Error>> + Send + 'static,
+    T: serde::de::DeserializeOwned + Send + 'static,
+{
+    Ok(tokio::task::spawn_blocking(move || {
+        let stream = pin!(stream);
+        let stream = stream.map_err(|e| io::Error::new(io::ErrorKind::Other, format!("{e:?}")));
+        let reader = SyncIoBridge::new(StreamReader::new(stream));
+        info_span!("parse document").in_scope(|| serde_yml::from_reader(reader))
```

Comment:
I don't love that both `*_from_stream` fn's are identical except for this one line. We can refactor later, I suppose, but it'd be nice to remove all the redundancy if you can.

Reply:
De-duplicated.